client/dashboard/sites/settings-sftp-ssh/ssh-card.tsx

```diff
@@ -0,0 +1,213 @@
+						</VStack>
+						<CoreBadge intent="info" style={ { height: '24px' } }>
+							{ sprintf(
+								/* translators: %s is when the SSH key was attached. */
+								__( 'Attached on %s' ),
+								new Intl.DateTimeFormat( userLocale, {
+									dateStyle: 'long',
+									timeStyle: 'medium',
+								} ).format( new Date( siteSshKey.attached_at ) )
+							) }
+						</CoreBadge>
+					</VStack>
+					<Button
+						icon={ trash }
+						label={ __( 'Detach' ) }
+						isBusy={ isBusy }
+						style={ { margin: '-6px' } }
+						onClick={ () => onDetach( siteSshKey ) }
+					/>
+				</HStack>
+			</CardBody>
+		</Card>
+	);
+};
+
+export default function SshCard( {
+	siteSlug,
+	sftpUsers,
+	sshEnabled,
+}: {
+	siteSlug: string;
+	sftpUsers: SftpUser[];
+	sshEnabled: boolean;
+} ) {
+	const [ selectedSshKey, setSelectedSshKey ] = useState( 'default' );
+	const { user } = useAuth();
+	const { data: siteSshKeys } = useQuery( siteSshKeysQuery( siteSlug ) );
+	const { data: profileSshKeys } = useQuery( profileSshKeysQuery() );
+	const toggleSshAccessMutation = useMutation(
+		! sshEnabled
+			? siteSshAccessEnableMutation( siteSlug )
+			: siteSshAccessDisableMutation( siteSlug )
+	);
+	const attachSshKeyMutation = useMutation( siteSshKeysAttachMutation( siteSlug ) );
+	const detachSshKeyMutation = useMutation( siteSshKeysDetachMutation( siteSlug ) );
+	const userLocale = user.locale_variant || user.language || 'en';
+	const { username } = sftpUsers[ 0 ];
+
+	const userKeyIsAttached = useMemo( () => {
+		if ( ! siteSshKeys ) {
+			return false;
+		}
+		return !! siteSshKeys.find( ( { user_login }: SiteSshKey ) => user_login === user.username );
+	}, [ siteSshKeys, user.username ] );
+
+	const showSshKeysSelect = ! userKeyIsAttached && profileSshKeys && profileSshKeys.length > 0;
+
+	const handleToggleSshAccess = () => {
+		toggleSshAccessMutation.mutate();
+	};
+
+	const handleAttachSshKey = () => {
+		attachSshKeyMutation.mutate( selectedSshKey );
+	};
+
+	const handleDetachSshKey = ( siteSshKey: SiteSshKey ) => {
+		detachSshKeyMutation.mutate( siteSshKey );
+	};
+
+	const handleSelectedSshKeyChange = ( currentSelectedKey: string ) => {
+		setSelectedSshKey( currentSelectedKey );
+	};
+
+	return (
+		<Card>
+			<CardBody>
+				<VStack style={ { paddingBottom: '12px' } }>
+					<Text size="15px" weight={ 500 } lineHeight="32px">
+						{ __( 'SSH' ) }
+					</Text>
+					<Text variant="muted" as="p">
+						{ createInterpolateElement(
+							__(
+								"SSH lets you access your site's backend via a terminal, so you can manage files and use <wpCliLink>WP-CLI</wpCliLink> for quick changes and troubleshooting. <learnMoreLink>Learn more</learnMoreLink>."
+							),
+							{
+								wpCliLink: <ExternalLink href="#" children={ null } />,
+								learnMoreLink: <ExternalLink href="#hosting-connect-to-ssh" children={ null } />,
+							}
+						) }
+					</Text>
+				</VStack>
+				<VStack spacing={ 4 } style={ { padding: '8px 0' } }>
```

Comment:
There are 3 actions here
* toggle the ssh access
* attach ssh key
* detach ssh key

Do we need to apply DataForm here 🤔